js/src/forum/components/IndexPage.js

```diff
@@ -33,26 +33,25 @@ export default class IndexPage extends Page {
     // probably want to refresh the results. We will clear the discussion list
     // cache so that results are reloaded.
     if (app.previous instanceof IndexPage) {
-      app.cache.discussionList = null;
+      app.cache.discussionList.clear();
     }
 
     const params = this.params();
 
-    if (app.cache.discussionList) {
+    if (app.cache.discussionList.hasDiscussions()) {
       // Compare the requested parameters (sort, search query) to the ones that
       // are currently present in the cached discussion list. If they differ, we
-      // will clear the cache and set up a new discussion list component with
-      // the new parameters.
+      // will clear the cache update the parameters.
       Object.keys(params).some((key) => {
-        if (app.cache.discussionList.props.params[key] !== params[key]) {
-          app.cache.discussionList = null;
+        if (app.cache.discussionList.getParams()[key] !== params[key]) {
+          app.cache.discussionList.clear();
           return true;
         }
       });
     }
 
-    if (!app.cache.discussionList) {
-      app.cache.discussionList = new DiscussionList({ params });
+    if (!app.cache.discussionList.hasDiscussions()) {
+      app.cache.discussionList.refresh(true, params);
```

Comment:
Not a big fan of boolean arguments - they hide intent at the call sites (e.g. here). What do you think? Typical alternatives:
- option arguments
- two methods with more explicit names (one can call each other in the implementation, obviously)

Reply:
Does _option arguments_ mean merge the two parameters into an options object, which would identify the options better by the object keys?

Reply:
We initially had 2 methods, but setParams isn't really ever used independently. I'd prefer an options object if it's down to those 2 choices.

Reply:
Not necessarily both arguments in the options hash, it could also be `refresh(params, {clear: true})`.